detect.go

```diff
@@ -108,26 +112,10 @@ func Detect() packit.DetectFunc {
 
 		httpdFpmPlan.Requires = append(baseRequirements, httpdFpmPlan.Requires...)
 		nginxFpmPlan.Requires = append(baseRequirements, nginxFpmPlan.Requires...)
-
-		plans := []packit.BuildPlan{httpdFpmPlan, nginxFpmPlan}
+		httpdFpmPlan.Or = []packit.BuildPlan{nginxFpmPlan}
 
 		return packit.DetectResult{
-			Plan: or(plans...),
+			Plan: httpdFpmPlan,
 		}, nil
 	}
```

Comment:
this change is more confusing to me than the code we had before.

Reply:
Fair. This is a shortcut to achieve the same result. The tests clearly require the structure where the `httpdFpmPlan` is the root plan with an `Or: nginxFpmPlan`.

I suspect there's some context on why the spec is the way it is due to backwards compatibility, but I really don't know. But the code as-written delivers the expected plan structure and it's somewhat simpler than before so I think it's useful.

Reply:
@sophiewigmore does this block merging?

Reply:
Yea, I'd prefer to put it back to the way it was before, which to me was more readable.

Reply:
@sophiewigmore look now?